src/hotspot/share/gc/g1/g1CollectionSetChooser.cpp

```diff
@@ -153,11 +152,13 @@ class G1BuildCandidateRegionsTask : public WorkerTask {
       // before we fill them up).
       if (should_add(r) && !G1CollectedHeap::heap()->is_old_gc_alloc_region(r)) {
```

Comment:
Pre-existing: why not move the `!G1CollectedHeap::heap()->is_old_gc_alloc_region(r)` to `should_add` helper?

Reply:
I already discussed this with @albertnetymk with the result of moving this to a separate commit.